src/ctapipe/containers.py

```diff
@@ -1388,43 +1322,88 @@ class SimulatedShowerDistribution(Container):
+        default_factory=TelescopeEventIndexContainer,
+        description="event indexing information",
+    )
+
+    simulation = Field(
+        None,
+        description="Simulated Event Information",
+        type=SimulationTelescopeContainer,
     )
-    r0 = Field(default_factory=R0Container, description="Raw Data")
-    r1 = Field(default_factory=R1Container, description="R1 Calibrated Data")
+
+    r0 = Field(default_factory=R0TelescopeContainer, description="Raw Data")
+    r1 = Field(default_factory=R1TelescopeContainer, description="R1 Calibrated Data")
     dl0 = Field(
-        default_factory=DL0Container, description="DL0 Data Volume Reduced Data"
+        default_factory=DL0TelescopeContainer,
+        description="DL0 Data Volume Reduced Data",
     )
-    dl1 = Field(default_factory=DL1Container, description="DL1 Calibrated image")
-    dl2 = Field(default_factory=DL2Container, description="DL2 reconstruction info")
-    simulation = Field(
-        None, description="Simulated Event Information", type=SimulatedEventContainer
+    dl1 = Field(
+        default_factory=DL1TelescopeContainer, description="DL1 images and parameters"
     )
-    trigger = Field(
-        default_factory=TriggerContainer, description="central trigger information"
+    dl2 = Field(
+        default_factory=DL2TelescopeContainer,
+        description="DL2 reconstructed event properties",
     )
     count = Field(0, description="number of events processed")
     pointing = Field(
-        default_factory=PointingContainer,
+        default_factory=TelescopePointingContainer,
         description="Array and telescope pointing positions",
     )
     calibration = Field(
-        default_factory=EventCalibrationContainer,
+        default_factory=TelescopeCalibrationContainer,
         description="Container for calibration coefficients for the current event",
     )
     mon = Field(
-        default_factory=MonitoringContainer,
+        default_factory=TelescopeMonitoringContainer,
         description="container for event-wise monitoring data (MON)",
     )
     muon = Field(
         default_factory=MuonContainer, description="Container for muon analysis results"
     )
 
 
+class SubarrayEventContainer(Container):
+    """Top-level container for all event information"""
+
+    count = Field(0, description="number of events processed")
+
+    index = Field(
+        default_factory=SubarrayEventIndexContainer,
+        description="event indexing information",
+    )
+
+    simulation = Field(
+        None,
+        description="Simulated Event Information",
+        type=SimulationSubarrayContainer,
+    )
+
+    dl0 = Field(
+        default_factory=DL0SubarrayContainer,
+        description="DL0 subarray wide information",
+    )
+
+    dl2 = Field(
+        default_factory=DL2SubarrayContainer,
+        description="DL2 reconstructed event properties",
+    )
+
+    pointing = Field(
```

Comment:
Move into dl0?

Reply:
There is also pointing information in the `tel` field below, although it is per telescope rather than for the subarray. 

Putting the subarray pointing info into the dl0 now seems strange, are there any quantities that depend on the subarray pointing at dl0? Isn't the telescope pointing the natural pointing until you do stereo?